### PR TITLE
Fix dry/moist thermo state diagnostics

### LINKED ISSUE #795: Incorrect diagnostic of conditional variables in quadratures
The current implementation of the environmental quadratures is used to diagnose dry and moist variables by sampling the dry/cloudy quadrature points. The information is then used in the buoyancy gradient diagnostics, and will be coupled with radiation in the future.

At the moment we were computing conditional averages as

x_sat_contrib = ∫ x(q,θ)*H(ql(q,θ)>0) ds,

which is the contribution of the saturated parcel to the environment (over which we integrate). This equation is the correct quadrature for environmental properties with exclusive contributions from saturated parcels, like the cloud fraction or the liquid water specific humidity. However, if we are interested in the value of the variables within the saturated volume, we need to divide by the cloud fraction,

x_sat = [ ∫ x(q,θ)*H(ql(q,θ)>0) ds ]  / [ ∫ H(ql(q,θ)>0) ds ]

PD: Error found by @trontrytel

### PR DESCRIPTION
**Pull request by ilopezgp**
Closes #795 .

### CODE REVIEW

**Review by charleskawczynski — APPROVED** [review]
Code changes look fine to me.

File: src/EDMF_Environment.jl [hunk 1]
```diff
@@ -231,21 +231,30 @@ function microphysics(en_thermo::SGSQuadrature, grid::Grid, state::State, precip
...
-            aux_en_sat.T[k] = outer_env[i_T_sat]
-            aux_en_sat.q_vap[k] = outer_env[i_qt_sat] - outer_env[i_ql]
-            aux_en_sat.q_tot[k] = outer_env[i_qt_sat]
-            ts_sat = TD.PhaseEquil_pTq(param_set, p0_c[k], aux_en_sat.T[k], aux_en_sat.q_tot[k])
-            aux_en_sat.θ_dry[k] = TD.dry_pottemp(ts_sat)
-            aux_en_sat.θ_liq_ice[k] = TD.liquid_ice_pottemp(ts_sat)
+            if aux_en.cloud_fraction[k] > 0
+                aux_en_sat.T[k] = outer_env[i_T_sat] / aux_en.cloud_fraction[k]
+                aux_en_sat.q_tot[k] = outer_env[i_qt_sat] / aux_en.cloud_fraction[k]
+                aux_en_sat.q_vap[k] = (outer_env[i_qt_sat] - outer_env[i_ql]) / aux_en.cloud_fraction[k]
```
Comment (trontrytel) [inline src/EDMF_Environment.jl:247]:
```suggestion
                aux_en_sat.q_vap[k] = (outer_env[i_qt_sat] - outer_env[i_ql] - outer_env[i_qi]) / aux_en.cloud_fraction[k]
```

**Review by trontrytel — COMMENTED** [review]
Should we also change the weights in lines 190 - 196?

File: src/EDMF_Environment.jl (hunk 1 above)
Comment (trontrytel) [reply src/EDMF_Environment.jl:247]: Or we could also just use the TD function: https://github.com/CliMA/Thermodynamics.jl/blob/5945e684e36d7759088097d00fe4562ceb9db0cf/src/relations.jl#L210

**Review by yairchn — APPROVED** [review]
LGTM

**Comment by ilopezgp** [comment]
> Should we also change the weights in lines 190 - 196?

I prefer not to change the weights, which are the weights for Gauss Hermite quadratures over the environment. When we diagnose conditional variables, we just have to divide by the cloud fraction, as discussed in #795 .

**Comment by trontrytel** [comment]
> > Should we also change the weights in lines 190 - 196?
> 
> I prefer not to change the weights, which are the weights for Gauss Hermite quadratures over the environment. When we diagnose conditional variables, we just have to divide by the cloud fraction, as discussed in #795 .

What I meant is that in those lines we also compute the conditional saturated vs unsaturated statistics:
https://github.com/CliMA/TurbulenceConvection.jl/blob/80947ac1f4c75c98d4f476a4ae1a9c72a68a6641/src/EDMF_Environment.jl#L190-L196

**Comment by ilopezgp** [comment]
> > > Should we also change the weights in lines 190 - 196?
> > 
> > 
> > I prefer not to change the weights, which are the weights for Gauss Hermite quadratures over the environment. When we diagnose conditional variables, we just have to divide by the cloud fraction, as discussed in #795 .
> 
> What I meant is that in those lines we also compute the conditional saturated vs unsaturated statistics:
> 
> https://github.com/CliMA/TurbulenceConvection.jl/blob/80947ac1f4c75c98d4f476a4ae1a9c72a68a6641/src/EDMF_Environment.jl#L190-L196

The division only has to be done once, I have done it after aggregation, which is when we have the cloud fraction available. It could be done here if we looped before just to get the cloud fraction, but this is more efficient.

**Pull request merged**